File: csharp/allergies/Allergies.cs
using System;
using System.Collections.Generic;
using System.Linq;

[Flags]
public enum Allergen : uint
{
    Eggs = 1,
    Peanuts = 2,
    Shellfish = 4,
    Strawberries = 8,
    Tomatoes = 16,
    Chocolate = 32,
    Pollen = 64,
    Cats = 128
}

public class Allergies
{
    private Allergen _mask { get; set; }

    public Allergies(int mask) => _mask = (Allergen)mask;

    // determine if person has allergies
    public bool IsAllergicTo(Allergen allergen) => _mask.HasFlag(allergen);

    // List of allergens to which person is allergic
    public IEnumerable<Allergen> List() => Enum.GetValues(typeof(Allergen)).Cast<Allergen>().Where(IsAllergicTo);
}

File: dart/allergies/lib/allergies.dart
class Allergies {
  Map<String, int> Allergens = {
    'eggs': 1,
    'peanuts': 2,
    'shellfish': 4,
    'strawberries': 8,
    'tomatoes': 16,
    'chocolate': 32,
    'pollen': 64,
    'cats': 128
  };

  bool allergicTo(String allergen, int score) {
    var n = Allergens[allergen];
    return n & score > 0;
  }

  List<String> list(int score) {
    return Allergens.keys.where((k) => allergicTo(k, score)).toList();
  }
}
